### PR TITLE
Add APIScan to notice build

### PR DESCRIPTION
**Pull request by weshaggard**
Part of issue https://github.com/Azure/azure-sdk-tools/issues/7085

### CODE REVIEW

File: eng/pipelines/build-notice.yml [hunk 1]
```diff
@@ -78,6 +78,36 @@ jobs:
...
       - task: ComponentGovernanceComponentDetection@0
 
+      - task: securedevelopmentteam.vss-secure-development-tools.build-task-apiscan.APIScan@2
+        displayName: 'Run APIScan'
+        condition: and(succeeded(), eq(variables['OSVmImage'], 'windows'))
+        inputs:
+          softwareFolder: $(Build.ArtifactStagingDirectory)
+          # SoftwareName cannot change as it is used for looking up baselining data
+          softwareName: 'Azure Developer CLI'
+          softwareVersionNum: '$(CLI_VERSION)'
```
Comment (danieljurek) [inline eng/pipelines/build-notice.yml:88]: This version number is the "expected next release" version number... Should this include other unique identifiers for the build?

Comment (weshaggard) [reply eng/pipelines/build-notice.yml:88]: I don't think they are particular for this version number.

**Comment by danieljurek** [comment]
Had a couple questions offline before approving.

**Comment by weshaggard** [comment]
APIScan doesn't correctly scan statically linked go exe's so closing this PR.

**Pull request closed without merging**